account_payment_order_notification/wizard/wizard_account_payment_order_notification.py

```diff
@@ -30,7 +30,11 @@ def default_get(self, fields):
         )
         if po:
             line_ids = []
-            for partner in po.payment_line_ids.mapped("partner_id"):
+            # We need to transfer invoice partner so that the email is sent to the
+            # correct partner (not parent)
+            for partner in po.payment_line_ids.mapped(
+                "move_line_id.move_id.partner_id"
```

Comment:
This is not correct: The move line may not be an invoice and don't have `partner_id`. You have to select the filled one (through a lambda in the mapped).

Reply:
Changes done (I believe both use cases are taken into account).